pmlc/dialect/stdx/transforms/i1_to_i32.cc

```diff
@@ -93,11 +104,132 @@ void populateI1StorageToI32(MLIRContext *context,
   patterns.insert<LoadOpI1ToI32, StoreOpI1ToI32>(context);
 }
 
+// Adaptor for building loop nests for the specific memRef shape
+void buildLoopForMemRef(
+    OpBuilder &builder, Location loc, Value memRef,
+    function_ref<void(OpBuilder &, Location, ValueRange)> bodyBuilder) {
+  auto memRefType = memRef.getType().dyn_cast_or_null<MemRefType>();
+  if (!memRefType)
+    return;
+
+  mlir::edsc::ScopedContext context(builder, loc);
```

Comment:
Is the context used anywhere?

Reply:
It is from EDSC declarative builders API.  Please see https://github.com/llvm/llvm-project/blob/master/mlir/docs/EDSC.md
Here we implemented memory copy in standard dialect instead of scf.for to avoid converting it to a gpu module.